synthesizer/process/src/trace/mod.rs

```diff
@@ -315,8 +315,8 @@ impl<N: Network> Trace<N> {
         }
         // Verify the proof.
         match VerifyingKey::verify_batch(locator, verifier_inputs, proof) {
-            true => Ok(()),
-            false => bail!("Failed to verify proof"),
+            Ok(()) => Ok(()),
+            Err(_) => bail!("Failed to verify proof"),
```

Comment:
it might be preferable to not ignore the reason for failing here; knowing if the proof was invalid or the inputs were duplicate could aid debugging

Reply:
Indeed, though the original code didn't propagate the error either, and there is [already a println! where it originates](https://github.com/AleoHQ/snarkVM/pull/2276/files#diff-4536d0528c90ad243676c73b67ffc8ae30cb325065825c168d5e48ab34ff2717R94). 

It may or may not be because we prefer to e.g. not return raw errors to clients. Let's see what Howard prefers.

Reply:
that [println](https://github.com/AleoHQ/snarkVM/pull/2276/files#diff-4536d0528c90ad243676c73b67ffc8ae30cb325065825c168d5e48ab34ff2717R94) is `cfg` guarded for CLI outputs only.

Reply:
see above examples - errors are always passed through

Reply:
resolved in c1d0898b5